src/composables/useCurrentTime.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { createSharedComposable } from '@vueuse/core'
+import { onUnmounted, readonly, ref } from 'vue'
+import type { DeepReadonly, Ref } from 'vue'
+
+/**
+ * Composable to get current time (as Date object)
+ * @param [precision] - precision in milliseconds (defaults to one minute)
+ * @return Date reactive object with current time
+ */
+export function useCurrentTimeComposable(precision: number = 60_000): DeepReadonly<Ref<Date>> {
+	let timeout: NodeJS.Timeout = null
+	let interval: NodeJS.Timeout = null
+
+	const date = ref<Date>(new Date())
+
+	timeout = setTimeout(() => {
+		date.value = new Date()
+		interval = setInterval(() => {
+			date.value = new Date()
+		}, precision)
+	}, precision - date.value % precision)
```

Comment:
`precision - date.value % precision` works only for a short time after composable was used, but will break after some time due to cumulative error.

Reply:
Interval is started on `X hours : Y min : 00 sec`. And it calls the function every N seconds, where N > 60.

If it would be `N === 60`, then the minute would update every actual minute change.

But `N > 60`. So we have updates on:
- 12:01:00
- 12:02:01
- 12:03:05
- 12:04:15
- ...
- 15:20:45
- 15:21:50
- 15:23:10

Reply:
Solution - only use `setTimeout`

Reply:
tested with 5-seconds precision
```js
console.log('requestUpdate', date.value.valueOf(), precision - date.value.valueOf() % precision)
<timestamp_seconds_milliseconds> <till next update>
 1747402_705_740 4260
 1747402_710_742 4258
 1747402_715_731 4269
 -- here tab is suspended --
 1747402_734_730 270
 1747402_794_733 267
 1747402_854_727 273
 -- back to normal --
 1747402_900_888 4112
 1747402_905_009 4991
 1747402_910_011 4989
```